tooling/azure-automation/resources-cleanup/src/automationassets.py

```diff
@@ -0,0 +1,147 @@
+""" Azure Automation assets module to be used with Azure Automation during offline development """
+#!/usr/bin/env python3
+# ----------------------------------------------------------------------------------
+#
+# MIT License
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+# ----------------------------------------------------------------------------------
+
+# Constant keys for extracing items from automation assets.
+_KEY_NAME = "Name"
+_KEY_VALUE = "Value"
+_KEY_USERNAME = "Username"
+_KEY_PASSWORD = "Password"
+_KEY_CERTPATH = "CertPath"
+_KEY_CONNECTION_FIELDS = "ValueFields"
+
+# Assets supported in Azure automation within python scripts
+_KEY_VARIABLE = "Variable"
+_KEY_CERTIFICATE = "Certificate"
+_KEY_CREDENTIAL = "Credential"
+_KEY_CONNECTION = "Connection"
+
+# Get Azure Automation asset json file
+def _get_automation_asset_file():
+    import os
+    if os.environ.get('AUTOMATION_ASSET_FILE') is not None:
+        return os.environ.get('AUTOMATION_ASSET_FILE')
+    return os.path.join(os.path.dirname(__file__), "localassets.json")
+
+# Helper function to find an asset of a specific type and name in the asset file
+def _get_asset_value(asset_file, asset_type, asset_name):
+    import json
+    json_data = open(asset_file)
+    json_string = json_data.read()
+    local_assets = json.loads(json_string)
+
+    return_value = None
+    for asset, asset_values in local_assets.iteritems():
+        if asset == asset_type:
+            for value in asset_values:
+                if value[_KEY_NAME] == asset_name:
+                    return_value = value
+                    break
+        if return_value != None:
+            # Found the value so break out of loop
+            break
+
+    return return_value
+
+# Returns an asset from the asses file
```

Comment:
```suggestion
# Returns an asset from the assets file
```

Reply:
Yes, this file is copy/pasted as per Azure Docs to be able to use locally variables defined in Azure automation (in the Cloud), similar to using env vars.

Reply:
I think we should/can still fix this comment and use local variables.

Reply:
Comment fixed.

Reply:
I still see the typo, but this can get addressed in a new PR, no need to block merging this.

Reply:
Ups, I forgot to git add. Done. Tks!